### PR TITLE
feat(layout): maximize window

### LINKED ISSUE #1641: maximizing splits
I would like to be able to instantly minimize / maximize splits using Ctrl-W o similarly to ZoomVim extension of the vim behavior:

https://github.com/vim-scripts/ZoomWin

any way to get that? :)

### PR DESCRIPTION
**Pull request by glennsl**
max2

This adds the ability to maximize a window, either vertically, horizontally or both, and to toggle/revert it back to the previous layout.

Fixes #1641

### CODE REVIEW

**Comment by glennsl** [comment]
The one hour time limit on Ubuntu seems a bit short @bryphe. It cancelled just 31s into the third round of integration tests.

**Comment by glennsl** [comment]
Cancelled again on Ubuntu. This time 50s into the first integration test.

**Comment by bryphe** [comment]
Bumped up the Ubuntu time limit to 90min

File: src/Feature/Layout/Layout.re [hunk 1]
```diff
@@ -810,3 +814,163 @@ let rec resizeSplit = (~path, ~delta, node) => {
...
+       actual == vsplit([window(1), window(~weight=10., 2)]);
+     };
+
+     let%test "vsplit - vertical" = {
+       let initial = vsplit([window(1), window(2)]);
+
+       let actual = maximize(~direction=`Vertical, 2, initial);
+
+       actual == vsplit([window(1), window(2)]);
+     };
```
Comment (bryphe) [inline src/Feature/Layout/Layout.re:885]: Nice set of test cases 👍

File: src/Store/KeyBindingsStoreConnector.re [hunk 2]
```diff
@@ -442,6 +442,23 @@ let start = maybeKeyBindingsFilePath => {
         command: Feature_Layout.Commands.resetSizes.id,
         condition: windowCommandCondition,
       },
+      // TODO: Fails to parse
+      // {
+      //   key: "<C-W>_",
```
Comment (bryphe) [inline src/Store/KeyBindingsStoreConnector.re:447]: Thanks for noting these!

File: src/Store/KeyBindingsStoreConnector.re [hunk 3]
```diff
@@ -442,6 +442,23 @@ let start = maybeKeyBindingsFilePath => {
...
+      //   condition: windowCommandCondition,
+      // },
+      // TODO: Fails to parse
+      // {
+      //   key: "<C-W>|",
+      //   command: Feature_Layout.Commands.maximizeHorizontal.id,
+      //   condition: windowCommandCondition,
+      // },
+      {
+        key: "<C-W>o",
```
Comment (bryphe) [inline src/Store/KeyBindingsStoreConnector.re:458]: This new keybinding is very nice

**Pull request merged**